File: wwwrootsrc/src/app/chat-message.ts
export class ChatMessage {
  public id: string;
  public message: string;
  public userName: string;
  public createdOn: Date;
  public updatedOn: Date;

  constructor(options) {
    this.id        = options.Id;
    this.message   = options.Message;
    this.userName  = options.UserName;
    this.createdOn = options.CreatedOn;
    this.updatedOn = options.UpdatedOn;
  }
}

File: src/server/Models/ChatMessage.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ChatApi.Models
{
    public class ChatMessage
    {
        [BsonId]
        [BsonIgnoreIfNullAttribute]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string Message { get; set; } = String.Empty;
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public DateTime UpdatedOn { get; set; } = DateTime.Now;
        public string UserName { get; set; } = String.Empty;
    }
}